src/phoenix/trace/otel/utils.py

```diff
@@ -0,0 +1,372 @@
+        return any_value.bytes_value
+    if which is None:
+        return None
+    assert_never(which)
+
+
+StatusMessage: TypeAlias = str
+
+_STATUS_DECODING = MappingProxyType(
+    {
+        otlp.Status.StatusCode.STATUS_CODE_UNSET: SpanStatusCode.UNSET,
+        otlp.Status.StatusCode.STATUS_CODE_OK: SpanStatusCode.OK,
+        otlp.Status.StatusCode.STATUS_CODE_ERROR: SpanStatusCode.ERROR,
+    }
+)
+
+
+def _decode_status(otlp_status: otlp.Status) -> Tuple[SpanStatusCode, StatusMessage]:
+    status_code = _STATUS_DECODING.get(otlp_status.code, SpanStatusCode.UNSET)
+    return status_code, otlp_status.message
+
+
+def _extract_sub_key(key: str, prefix: str) -> Optional[str]:
+    prefix_dot = f"{prefix}."
+    if not (len(prefix_dot) < len(key) and key.startswith(prefix_dot)):
+        return None
+    return key[len(prefix_dot) :]
+
+
+def _extract_index_and_sub_key(key: str, prefix: str) -> Optional[Tuple[int, str]]:
+    indexed_sub_key = _extract_sub_key(key, prefix)
+    if not indexed_sub_key:
+        return None
+    dot_idx = indexed_sub_key.find(".")
+    if not (0 < dot_idx < len(indexed_sub_key) - 1):
+        return None
+    index_prefix = indexed_sub_key[:dot_idx]
+    if not index_prefix.isdigit():
+        return None
+    index = int(index_prefix)
+    sub_key = indexed_sub_key[dot_idx + 1 :]
+    return index, sub_key
+
+
+def _consolidate_prefixed_indexed_keys_into_list(
+    attributes: Mapping[str, Any],
+    prefix: str,
+) -> Tuple[Optional[List[Dict[str, Any]]], Optional[List[str]]]:
+    """Consolidate keys with the given prefix into a single list (of dictionaries).
+    Return the consolidated list and the list of keys that were consolidated."""
+    relevant_keys = [
+        (key, idx_and_sub_key, value)
+        for key, value in attributes.items()
+        if (idx_and_sub_key := _extract_index_and_sub_key(key, prefix)) is not None
+    ]
+    if not relevant_keys:
+        return None, None
+    indexed: DefaultDict[int, Dict[str, Any]] = defaultdict(dict)
+    for _, (idx, sub_key), value in relevant_keys:
+        indexed[idx][sub_key] = value
+    return [dictionary for _, dictionary in sorted(indexed.items())], [
+        key for key, *_ in relevant_keys
+    ]
+
+
+def _consolidate_prefixed_keys_into_dict(
+    attributes: Mapping[str, Any],
+    prefix: str,
+) -> Tuple[Optional[Dict[str, Any]], Optional[List[str]]]:
+    """Consolidate keys with the given prefix into a single dictionary.
+    Return the consolidated dictionary and the list of keys that were consolidated."""
+    relevant_keys = [
+        (key, sub_key, value)
+        for key, value in attributes.items()
+        if (sub_key := _extract_sub_key(key, prefix)) is not None
+    ]
+    if not relevant_keys:
+        return None, None
+    return {sub_key: value for _, sub_key, value in relevant_keys}, [
+        key for key, *_ in relevant_keys
+    ]
+
+
+_NANO = 1_000_000_000  # for converting seconds to nanoseconds
```

Comment:
Let's add units to the const

Reply:
this one actually doesn't have a unit because it's a conversion factor